File: server/Service/InputModels/ServerInputModel.cs
﻿namespace Service.InputModels;

using Domain.Enums;

public class ServerInputModel
{
    public ServerInputModel(string serverName, string serverType, string exactVersion, int maxMemory, ServerPropertiesInputModel? serverProperties)
    {
        this.serverName = serverName;
        this.serverType = serverType;
        this.exactVersion = exactVersion;
        this.maxMemory = maxMemory;
        this.serverProperties = serverProperties;
    }

    public string serverName { get; set; }
    public string serverFileName { get; set; }
    public string serverType { get; set; }
    public string exactVersion { get; set; }
    public int maxMemory { get; set; }
    public ServerPropertiesInputModel? serverProperties { get; set; }
}

File: client/src/app/core/models/serverInput.model.ts
import { ServerPropertiesInput } from "./serverPropertiesInput.model";

export class ServerInput {
  constructor(
    public serverName: string | null | undefined,
    public serverType: string | null | undefined,
    public exactVersion: string | null | undefined,
    public maxMemory: number | null | undefined,
    public serverFileName: string | null | undefined,
    public serverProperties: ServerPropertiesInput | null

  ) {}
}
